ert_data/loader.py

```diff
@@ -8,101 +8,215 @@ def data_loader_factory(observation_type):
         See https://github.com/equinor/libres/issues/808
     """
     if observation_type == "GEN_OBS":
-        return load_general_data
+        return _load_general_data
     elif observation_type == "SUMMARY_OBS":
-        return load_summary_data
+        return _load_summary_data
     elif observation_type == "BLOCK_OBS":
-        return load_block_data
+        return _load_block_data
     else:
         raise TypeError("Unknown observation type: {}".format(observation_type))
 
 
-def load_general_data(facade, observation_key, case_name, include_data=True):
-    obs_vector = facade.get_observations()[observation_key]
-    data_key = obs_vector.getDataKey()
+def _extract_data(
+    facade, obs_keys, case_name, response_loader, obs_loader, expected_obs, include_data
+):
+    if isinstance(obs_keys, str):
+        obs_keys = [obs_keys]
+    data_map = {}
+    obs_types = [facade.get_impl_type_name_for_obs_key(key) for key in obs_keys]
+    data_keys = [facade.get_data_key_for_obs_key(key) for key in obs_keys]
+    assert (
+        len(set(obs_types)) == 1
+    ), f"\nExpected only {expected_obs} observation type. Found: {obs_types} for {obs_keys}"
```

Comment:
Want to use `set(obs_types)` here?

Reply:
Yes? This is the first place this function is being called, and is only valid for one obs type at the time. Would you prefer creating an exception further down? We risk that exception being `util_abort`.

Reply:
As in unique `obs_types` in the error message :) Had a fear it could drown out a bit

That was really ambiguous!

Reply:
Ahh! I misunderstood completely, thought you had a question about the assert! I will fix it.

Reply:
Just remembered the reason I did it like this. It is because I wanted to map which key belonged to the wrong observation, which is why I opted to print both the lists in their entirety. Good enough argument, or just print what is wrong?

Reply:
Then the user would be presented with two (long?) lists where he/she would need to map the index of the incorrect type to the corresponding location in the other..? The expected observation type is included in the printing, so I suggest removing those that are correct and focus only on the incorrect ones

  `"/".join([f"key {k} has type {t}" for k,t in zip(obs_keys, obs_types) if t != expected_obs_type])`

Feel free to break my one-liner a bit :)

Reply:
That just feel like overkill to do for an error message, but might be warranted :p It might be long, sure, but even if it is, wont be _that_ hard will it? Just feel like adding that much logic to an error might cause another error ;)

Reply:
I'm not sure how this will hit, but if you have 1 out of 1000 observations be a different obs type (say it's the 348th) -> you'll end up with two lists of 1000 strings printed (majority identical), where you first need to find the deviate observation type, and once it's found you need to map it to the same index in the `obs_keys` list.

Not sure if 1000 is representative though, but anything more than 5 is likely to be error-prone and an annoyance.

So I figured, why print the accepted values in addition to the invalid ones?

Reply:
Well, we want to encourage people to use `MeasuredData`, not this as an entry point, so make it give all the feedback? ;)

Reply:
Cause this is already asserted in `measured_data.py:160 def _get_data`? which is the function that calls the factory anyway. The argument then is that we can be more verbose here1? (I think you've already explained me this, but why are we asserting twice the uniqueness of `obs_type`?)